src/main/java/executor/command/CommandConvert.java

```diff
@@ -0,0 +1,194 @@
+package executor.command;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.RoundingMode;
+import java.net.URL;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import executor.task.TaskList;
+import interpreter.Parser;
+import ui.Ui;
+import ui.Wallet;
+import java.math.BigDecimal;
+
+/**
+ * This command helps to convert user entered currency amount into another country's currency amount using api call.
+ */
+public class CommandConvert extends Command {
+    private String from = " ";
+    private String to = " ";
+    private Double amount;
+    private Double exchangeRate = 0.00;
+
+    /**
+     * Constructor for the CommandConvert class.
+     * @param userInput this is the userInput from the CLI
+     */
+    public CommandConvert(String userInput) {
+        this.userInput = userInput;
+        this.commandType = CommandType.CONVERT;
+        this.amount = extractAmount(this.commandType, userInput);
+        this.from = getFromUserInput(userInput);
+        this.to = getToUserInput(userInput);
+    }
+
+
+    /**
+     * execute is the overall function which calls the converter and ui.
+     * @param wallet wallet is the overall wallet object
+     */
+    @Override
+    public void execute(Wallet wallet) {
+        Double convertedAmount = this.convertCurrency(this.getFrom(), this.getTo(), this.getAmount());
+        Ui.dukeSays(this.result(convertedAmount) + rateUsed());
+        Ui.printSeparator();
+    }
+
+
+    @Override
+    public void execute(TaskList taskList) {
+
+    }
+
+
+    /**
+     * extractAmount parses the user input from CLI to get the amount which the user wishes to convert.
+     * @param commandType this is the type of command
+     * @param userInput takes the user entered input from CLI
+     * @return the amount which user wishes to convert is returned
+     */
+    private Double extractAmount(CommandType commandType, String userInput) {
+        String amountStr = Parser.parseForPrimaryInput(commandType, userInput);
+        return Double.parseDouble(amountStr);
+    }
+
+    /**
+     * getFromUserInput parses user input for the flag "from".
+     * @param userInput this is the user entered input from CLI
+     * @return this function returns the 3 character unique string
+     */
+    private String getFromUserInput(String userInput) {
```

Comment:
Consider refactoring method name to _getCurrencyConvertFrom_ for greater clarity.

Reply:
refactored and changed  the method name to getCurrencyConvertFrom